src/Shared/PackageActions/NuGetPackageActions.cs

```diff
@@ -134,12 +141,18 @@ public class NuGetPackageActions : IManagerPackageActions<NuGetPackageVersionMet
     {
         FindPackageByIdResource resource = await _sourceRepository.GetResourceAsync<FindPackageByIdResource>();
 
-        IEnumerable<NuGetVersion> versionsAscending = await resource.GetAllVersionsAsync(
+        ImmutableList<NuGetVersion> versionsAscending = (await resource.GetAllVersionsAsync(
             packageUrl.Name,
             _sourceCacheContext,
             _logger, 
-            cancellationToken);
+            cancellationToken)).ToImmutableList();
```

Comment:
Why this change to fully enumerate here?

Reply:
Removed.